tests/ConfigurationSchemaGenerator.Tests/GeneratorTests.cs

```diff
@@ -997,7 +999,144 @@ public record TestSettings2
     }
 
     [Fact]
-    public void PreservesExistingComponentTypeWhenSkipped()
+    public void MergesPropertiesAtConfigPathWithDifferentCasing()
+    {
+        var source =
+            """
+            [assembly: Aspire.ConfigurationSchema("One:Two:Three", typeof(TestSettings1))]
+            [assembly: Aspire.ConfigurationSchema("ONE:two:THREE", typeof(TestSettings2))]
+
+            public record TestSettings1
+            {
+                /// <summary>1</summary>
+                public string? TestProperty1 { get; set; }
+            }
+
+            public record TestSettings2
+            {
+                /// <summary>2</summary>
+                public int? TestProperty2 { get; set; }
+            }
+            """;
+
+        var schema = GenerateSchemaFromCode(source, []);
+
+        AssertIsJson(schema,
+            """
+            {
+              "type": "object",
+              "properties": {
+                "One": {
+                  "type": "object",
+                  "properties": {
+                    "Two": {
+                      "type": "object",
+                      "properties": {
+                        "Three": {
+                          "type": "object",
+                          "properties": {
+                            "TestProperty1": {
+                              "type": "string",
+                              "description": "1"
+                            },
+                            "TestProperty2": {
+                              "type": "integer",
+                              "description": "2"
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+            """);
+    }
+
+    [Fact]
+    public void MergesTypesWithDifferentCasing()
```

Comment:
I'm a bit confused by this test. The string `testSETTINGS` doesn't show up at all in the JSON. Should it?

Reply:
No, it shouldn't. The names of non-primitive types never show up in the schema. This was already the case and is the correct behavior.

What this PR changes is that the schema contains: `"description": "Replaced documentation."`. Without the changes in this PR, it results in: `"description": "Initial documentation."`. The test shows the last occurrence wins, which used to be the first occurrence (but there was no test for it).

As you indicated we should be consistent whether the first or last occurrence wins, I'm going to rename the tests to include this information. The new test names should make it easier to understand their purpose.